specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.4-preview.1/secrets.json

```diff
@@ -877,6 +1008,70 @@
         }
       },
       "description": "The backup secret result, containing the backup blob."
+    },
+    "SecretRotationPolicy": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The secret rotation policy id.",
+          "readOnly": true
+        },
+        "lifetimeActions": {
+          "type": "array",
+          "items": {
+            "$ref": "common.json#/definitions/LifetimeActions"
+          },
+          "description": "Actions that will be performed by Key Vault over the lifetime of a secret. For preview, lifetimeActions can only have two items at maximum: one for rotate, one for notify. Notification time would be default to 30 days before expiry and it is not configurable."
+        },
+        "attributes": {
+          "$ref": "#/definitions/SecretRotationPolicyAttributes",
+          "description": "The secret rotation policy attributes."
+        },
+        "autoRotationEnabled": {
+          "type": "boolean",
+          "description": "Secret auto rotation would be enabled if set to be true. Otherwise, it would be false"
+        },
+        "providerType": {
+          "type": "string",
+          "description": "String value which defines the used provider."
+        },
+        "providerSettings": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Provider settings in the form of key/value pairs."
+        },
+        "autoRotationSettings": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Secret autorotation settings in the form of key/value pairs."
+        }
+      },
```

Comment:
Before sure to document required properties. Same with any new model introduced here.